internal/ignition/installmanifests.go

```diff
@@ -800,11 +799,14 @@ func (g *installerGenerator) updateBootstrap(ctx context.Context, bootstrapPath
 	return nil
 }
 
-func popNextHost(hosts []*models.Host) (*models.Host, []*models.Host) {
-	if len(hosts) == 0 {
-		return nil, nil
+func removeHostFromSlice(hostname string, hosts []*models.Host) (*models.Host, []*models.Host) {
+	for i, host := range hosts {
+		if hostname == hostutil.GetHostnameForMsg(host) {
+			hosts = append(hosts[:i], hosts[i+1:]...)
+			return host, hosts
+		}
 	}
-	return hosts[0], hosts[1:]
+	return nil, nil
 }
```

Comment:
I'm not convinced this still matters, but if we're not willing to clean up the fallback case where hostnames don't match in `getBmhRole` then I think we should have a similar backwards compatibility change/fallback here.

I'd say keep the length check, then do your new loop, then keep this line as is.

Reply:
So something like:
```go
func removeHostFromSlice(hostname string, hosts []*models.Host) (*models.Host, []*models.Host) {
  if len(hosts) == 0 {
		return nil, nil
  }
  for i, host := range hosts {
		if hostname == hostutil.GetHostnameForMsg(host) {
			hosts = append(hosts[:i], hosts[i+1:]...)
			return host, hosts
		}
  }
  return hosts[0], hosts[1:]
}
```

Sorry for the formatting.

Reply:
Thanks for the suggestion. Updated.